### PR TITLE
feat: host runtime info structures to be handled correctly

### PR DESCRIPTION
**Pull request by ireznice**

### CODE REVIEW

File: delayed.py [hunk 1]
```diff
@@ -90,8 +105,15 @@ def host_info_obtainer(conn, vc):
 
     process_actions = True
     while process_actions:
+
+        with data.Connection.use('conn2') as conn:
+            try:
+                host_info_obtainer(conn, vc)
+            except Exception:
+                Settings.raven.captureException(exc_info=True)
+                logger.error('Exception while processing request: ', exc_info=True)
```
Comment (petrSchreiber) [inline delayed.py:114]: I would maybe propose:
```suggestion
                logger.error('Could not obtain host information: ', exc_info=True)
```

File: delayed.py [hunk 2]
```diff
@@ -73,8 +73,23 @@ def host_info_obtainer(conn, vc):
...
+                hosts_to_be_deleted.append(hostinfo.name)
+        for host_name in hosts_to_be_deleted:
+            host = data.HostRuntimeInfo.get({'name': host_name}, conn=conn)
+            ids_to_be_deleted.append(host[0].id)
+        for host_id in ids_to_be_deleted:
+            data.HostRuntimeInfo.delete({'_id': host_id}, conn=conn)
+            logger.debug(f'host: {host_id} deleted from database')
+
+        logger.info(f'host_info_obtainer finished successfully ' +
+                    f'in: {time.time() - start_host_info_obtainer}')
```
Comment (petrSchreiber) [inline delayed.py:92]: Apologies, but the function is called "host_info_obrainer" and this part of code performs data deletion - I would extract it to a separate method.

**Review by petrSchreiber — CHANGES_REQUESTED** [review]
I really appreciate improving the functionality, but I propose some minor changes.

File: delayed.py (hunk 1 above)
Comment (ireznice) [reply delayed.py:114]: OK

File: delayed.py (hunk 2 above)
Comment (ireznice) [reply delayed.py:92]: it may be done

Comment (ireznice) [reply delayed.py:92]: ok but array `info` should be returned from the original function and given to the next function in the chain as a param?

Comment (ireznice) [reply delayed.py:92]: I've added something but result from current code is needed!

**Review by petrSchreiber — APPROVED** [review]
Thank you very much, Ivo. I appreciate the fine tuning 🙏

**Pull request merged**